plugins/module_utils/import_handler.py

```diff
@@ -27,6 +27,48 @@ def method(*args, **kwargs):
         return method
 
 
+class ZOAUImportError(object):
+    """This class serves as a wrapper for any kind of error when importing
+    ZOAU. Since ZOAU is used by both modules and module_utils, we need a way
+    to alert the user when they're trying to use a function that couldn't be
+    imported properly. If we only had to deal with this in modules, we could
+    just validate that imports worked at the start of their main functions,
+    but on utils, we don't have an entry point where we can validate this.
+    Just raising an exception when trying the import would be better, but that
+    introduces a failure on Ansible sanity tests, so we can't do it.
+
+    Instead, we'll replace what would've been a ZOAU library with this class,
+    and the moment ANY method gets called, we finally raise an exception.
+    """
+
+    def __init__(self, exception_traceback: str):
+        """When creating a new instance of this class, we save the traceback
+        from the original exception so that users have more context when their
+        task/code fails. The expected traceback is a string representation of
+        it, not an actual traceback object. By importing `traceback` from the
+        standard library and calling `traceback.format_exc()` we can
+        get this string.
+        """
+        self.traceback = exception_traceback
+
+    def __getattr__(self, name):
+        """This code is virtually the same from `MissingZOAUImport`. What we
+        do here is hijack all calls to any method from a missing ZOAU library
+        and instead return a method that will alert the user that there was
+        an error while importing ZOAU.
```

Comment:
I never stopped to read MissingZOAUImport, but reading through the original issue and your comment I assume just using import and then raise exception breaks the ansible sanity tests?

Reply:
Yes, sadly the most logical approach breaks sanity tests. Running Ansible's sanity tests with just raising an exception the moment we get an error importing ZOAU gives me this:

```
Running sanity test "import" on Python 3.10
ERROR: Found 4 import issue(s) on python 3.10 which need to be resolved:
ERROR: plugins/module_utils/job.py:36:0: traceback: ModuleNotFoundError: No module named 'zoautil_py'
ERROR: plugins/modules/zos_job_output.py:417:0: traceback: ModuleNotFoundError: No module named 'zoautil_py' (at plugins/module_utils/job.py:36:0)
ERROR: plugins/modules/zos_job_query.py:258:0: traceback: ModuleNotFoundError: No module named 'zoautil_py' (at plugins/module_utils/job.py:36:0)
ERROR: plugins/modules/zos_job_submit.py:610:0: traceback: ModuleNotFoundError: No module named 'zoautil_py' (at plugins/module_utils/job.py:36:0)
```

And this (the other 3 errors are about ignore files and one about doing things unidiomatically, so unrelated to your comment right now): 

```
Running sanity test "validate-modules"
ERROR: Found 6 validate-modules issue(s) which need to be resolved:
ERROR: plugins/modules/zos_job_output.py:0:0: import-error: Exception attempting to import module for argument_spec introspection, 'No module named 'zoautil_py''
ERROR: plugins/modules/zos_job_query.py:0:0: import-error: Exception attempting to import module for argument_spec introspection, 'No module named 'zoautil_py''
ERROR: plugins/modules/zos_job_submit.py:0:0: import-error: Exception attempting to import module for argument_spec introspection, 'No module named 'zoautil_py''
```